dev/scripts/src/contract/fundDapps.ts

```
@@ -0,0 +1,124 @@
+import { TransactionQueue } from '@prosopo/tx'
+import { defaultConfig } from '@prosopo/cli'
+import { getPairAsync, oneUnit } from '@prosopo/contract'
+import { hexToU8a } from '@polkadot/util'
+import { loadEnv } from '@prosopo/cli'
+import { setupDapp } from '../setup/index.js'
+import mongoose, { Model } from 'mongoose'
+import path from 'path'
+
+const log = getLogger(LogLevel.enum.info, 'dev.deploy')
+
+interface Emails {
+    email: string
+    name: string
+    url: string
+    account: string
+    mnemonic: string
+    createdAt: number
+    marketingPreferences: boolean
+}
+
+const EmailModelSchema = new mongoose.Schema<Emails>(
+    {
+        email: String,
+        name: String,
+        url: String,
+        account: String,
+        mnemonic: String,
+        createdAt: Number,
+        marketingPreferences: Boolean,
+    },
+    { collection: 'emails' }
+)
+
+let EmailsModel: typeof Model<Emails>
+try {
+    EmailsModel = mongoose.model('emails')
+} catch (error) {
+    EmailsModel = mongoose.model('emails', EmailModelSchema)
+}
+
+export async function run(wasmPath: string | undefined, abiPath: string | undefined, atlasUri?: string) {
+    // if wasmPath not provided then default to the captcha contract's wasm
+    let wasm: Uint8Array
+    if (wasmPath === undefined) {
+        log.info('Using wasm from captcha contract')
+        const jsonContent = JSON.parse(ContractFile)
+        const hex = jsonContent['source']['wasm']
+        wasm = hexToU8a(hex)
+    } else {
+        log.info('WASM Path', wasmPath)
+        wasm = await Wasm(path.resolve(wasmPath))
+    }
+    // if abiPath not provided then default to the captcha contract's abi
+    let abi: Abi
+    if (abiPath === undefined) {
+        log.info('Using abi from captcha contract')
+        abi = new Abi(ContractAbi)
+    } else {
+        log.info('ABI Path', abiPath)
+        abi = await AbiJSON(path.resolve(abiPath))
+    }
+
+    if (!atlasUri) {
```

Can we move this to the top of the func so it fails fast

---

Good spot. Definitely shouldn't be an optional param.